extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -21,7 +25,38 @@ import com.mapbox.maps.extension.style.utils.toValue
  * @see [The online documentation](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#geojson)
  *
  */
-class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
+class GeoJsonSource(
+  builder: Builder
+) : Source(builder.sourceId) {
+
+  var geoJsonParsed = false
+  var onGeoJsonParsedListener: ((GeoJsonSource) -> Unit)? = null
```

Comment:
can we abstract these two logic to some generic class?

Reply:
Taken another look and now I don't understand what you're talking about 😄 
We basically need a setter for `onGeoJsonParsedListener` because in `MapboxMap.onFinishLoadingStyleExtension` we need to set new listener if necessary. 
At the same time I don't want bringing in generic class because all other sources are sync and it's pointless right now adding those abstractions.